packages/fiori/cypress/specs/ViewSettingsDialog.cy.tsx

```diff
@@ -168,3 +169,197 @@ describe("View settings dialog - selection", () => {
+			<ViewSettingsDialog id="vsd">
+				<SortItem slot="sortItems" text="Name"></SortItem>
+				<SortItem slot="sortItems" text="Position"></SortItem>
+				<SortItem slot="sortItems" text="Company"></SortItem>
+				<SortItem slot="sortItems" text="Department"></SortItem>
+			</ViewSettingsDialog>
+		);
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
+			.shadow()
+			.find("ui5-list ui5-li[selected]")
+			.should("exist")
+			.should("contain.text", "Ascending");
+
+		cy.get("#vsd")
+			.shadow()
+			.find("[sort-by] ui5-li[selected]")
+			.should("not.exist");
+
+		cy.get("body").realPress("Escape");
+	});
+
+	it("should change sortOrder and confirm selection", () => {
+		cy.mount(
+			<ViewSettingsDialog id="vsd">
+				<SortItem slot="sortItems" text="Name"></SortItem>
+				<SortItem slot="sortItems" text="Position"></SortItem>
+				<SortItem slot="sortItems" text="Company"></SortItem>
+				<SortItem slot="sortItems" text="Department"></SortItem>
+			</ViewSettingsDialog>
+		);
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
+			.shadow()
+			.find("ui5-list ui5-li")
+			.eq(1)
+			.realClick();
+
+		cy.get("#vsd")
+			.shadow()
+			.find(".ui5-vsd-footer ui5-button")
+			.realClick();
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
+			.shadow()
+			.find("ui5-list ui5-li[selected]")
+			.should("contain.text", "Descending");
+
+		cy.get("body")
+			.realPress("Escape");
+	});
+
+	it("should change sortBy selection and verify persistence", () => {
+		cy.mount(
+			<ViewSettingsDialog id="vsd">
+				<SortItem slot="sortItems" text="Name"></SortItem>
+				<SortItem slot="sortItems" text="Position"></SortItem>
+				<SortItem slot="sortItems" text="Company"></SortItem>
+				<SortItem slot="sortItems" text="Department"></SortItem>
+			</ViewSettingsDialog>
+		);
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
+			.shadow()
+			.find("[sort-by] ui5-li")
+			.realClick();
+
+		cy.get("#vsd")
+			.shadow()
+			.find(".ui5-vsd-footer ui5-button")
+			.realClick();
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
+			.shadow()
+			.find("[sort-by] ui5-li[selected]")
+			.should("contain.text", "Name");
+
+		cy.get("body")
+			.realPress("Escape");
+	});
+
+	it("should reset settings to initial values", () => {
+		cy.mount(
+			<ViewSettingsDialog id="vsd">
+				<SortItem slot="sortItems" text="Name"></SortItem>
+				<SortItem slot="sortItems" text="Position"></SortItem>
+				<SortItem slot="sortItems" text="Company"></SortItem>
+				<SortItem slot="sortItems" text="Department"></SortItem>
+			</ViewSettingsDialog>
+		);
+
+		cy.get("[ui5-view-settings-dialog]")
+			.as("vsd")
+			.invoke("prop", "open", true);
+
+		cy.get("#vsd")
```

Comment:
you already have vsd stored as alias. it would suggest to use it instead of the id selector